include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -6636,12 +6636,12 @@ FEEvaluation<dim,
     }
 
 #  ifdef DEBUG
-  if (evaluation_flag_actual & EvaluationFlags::values)
-    this->values_quad_initialized = true;
-  if (evaluation_flag_actual & EvaluationFlags::gradients)
-    this->gradients_quad_initialized = true;
-  if (evaluation_flag_actual & EvaluationFlags::hessians)
-    this->hessians_quad_initialized = true;
+  this->values_quad_initialized =
+    evaluation_flag_actual & EvaluationFlags::values;
+  this->gradients_quad_initialized =
+    evaluation_flag_actual & EvaluationFlags::gradients;
+  this->hessians_quad_initialized =
+    evaluation_flag_actual & EvaluationFlags::hessians;
```

Comment:
The fix is to set the fields to whatever was requested at the particular calling function.